packages/react/src/components/input/input.tsx

```diff
@@ -1,21 +1,39 @@
 import { c, classy, InputProps as BaseProps, m } from '@onfido/castor';
 import React from 'react';
+import { FieldLabelWrapper } from '../../internal';
 import { withRef } from '../../utils';
 
+const idPrefix = 'castor_input';
+let idCount = 0;
+
 export const Input = withRef(
   (
-    { type = 'text', invalid, className, ...restProps }: InputProps,
+    {
+      id = `${idPrefix}_${idCount++}`,
+      type = 'text',
+      invalid,
+      children,
+      className,
+      ...restProps
+    }: InputProps,
     ref: InputProps['ref']
   ): JSX.Element => (
-    <input
-      {...restProps}
-      ref={ref}
-      type={type}
-      className={classy(c('input'), m({ invalid }), className)}
-    />
+    <FieldLabelWrapper id={id}>
+      {{
+        children,
+        element: (
+          <input
+            {...restProps}
+            ref={ref}
+            id={id}
```

Comment:
🤓  we include an `id` even if the input has no label.
Not a problem, just a bit unnecessary.

Reply:
The problem here is that I do allow to set `id` via the prop, then default to internal one. I suppose a check could be done to e.g. see if a prefix is there in the ID, but what if someone actually decides to use such as custom one? Not very likely, but still might introduce a bug?

Obviously, instead of defaulting `id` to internal one I could set it within component in case of label is used but no `id` is set via the prop, but that also includes a little more component logic.

Would you prefer this?

Reply:
I've refactored this part now, as there was also an issue with re-renders to generate ID each time.